File: leetcode/easy/28.Find the Index of the First Occurrence in a String/python/findTheIndexOfTheFirstOccurrenceOfAString.py
def strStr(haystack, needle):
    '''time complexity O(nm) Where n = lenght of haystack Where m = lenght of needle
       space complexity O(1)'''
    n = len(haystack)
    m = len(needle)
    for window_start in range(n - m + 1):
        for i in range(m):
            if needle[i] != haystack[window_start + i]:
                break
            if i == m - 1:
                return window_start
    return - 1

h = 'lietceet'
n = 'eet'
print(strStr(h, n))

File: leetcode/easy/28.Find the Index of the First Occurrence in a String/js/findTheIndexOfTheFirstOccurrenceOfAString.js
/**
 * @param {string} haystack
 * @param {string} needle
 * @return {number}
 * @author AlexSoto
 */

// time complexity O(nm)
// Where n = lenght of haystack
// Where m = length o needle
// space complexity O(1)
var strStr = function(haystack, needle) {
    let n = haystack.length;
    let m = needle.length;
    for (let windowStart = 0; windowStart <= n - m; windowStart++) {
        for (let i = 0; i < m; i++) {
            if (needle[i] !== haystack[windowStart + i]) {
                break;
            };
            if (i === m - 1) {
                return windowStart;
            };
        };
    };
    return -1
};

// Testing
h = "sadbutsad"
n = "sad"

console.log(strStr(h, n))